src/lib/utils.ts

```diff
@@ -0,0 +1,40 @@
+export function deepFreeze<T extends object>(target: T): T {
```

Comment:
let's name this file `deep-freeze.ts` and keep `isFunction` and `isObjectLike` functions private

Reply:
Done